src/Nethermind/Nethermind.Consensus/Processing/BlockProcessor.cs

```diff
@@ -65,8 +68,7 @@ public partial class BlockProcessor : IBlockProcessor
         _withdrawalProcessor = withdrawalProcessor ?? new WithdrawalProcessor(stateProvider, logManager);
         _rewardCalculator = rewardCalculator ?? throw new ArgumentNullException(nameof(rewardCalculator));
         _blockTransactionsExecutor = blockTransactionsExecutor ?? throw new ArgumentNullException(nameof(blockTransactionsExecutor));
-        _beaconBlockRootHandler = new BeaconBlockRootHandler();
-
+        _beaconBlockRootHandler = beaconBlockRootHandler ?? new BeaconBlockRootHandler(transactionProcessor, logManager);
```

Comment:
Always pass it here and don't create, then you don't have to pass transaction processor and it will become simpler.

Reply:
makes sense